dialogs/primitives.py

```diff
@@ -1,55 +1,68 @@
 from typing import List, Any
+from dataclasses import dataclass
 
 from .types import Dialog, ClientResponse, RunSubdialog, message, send_to_client
 
 
-def prompt(text) -> Dialog:
-    def _prompt(run: RunSubdialog) -> ClientResponse:
-        run(message(text))
+@dataclass(frozen=True)
+class prompt(Dialog):
+    text: str
+    version = "1.0"
+
+    def __call__(self, run: RunSubdialog) -> ClientResponse:
+        run(message(self.text))
         return run(send_to_client())
 
-    return _prompt
 
+@dataclass(frozen=True)
+class chain(Dialog):
+    dialogs: list
+    version = "1.0"
 
-def chain(dialogs: list) -> Dialog:
-    def _chain(run: RunSubdialog) -> List[Any]:
-        return [run(dialog) for dialog in dialogs]
+    def __call__(self, run: RunSubdialog) -> List[Any]:
```

Comment:
Should use generics

Reply:
Let's discuss, I'm not sure what the type should be